data_safe_haven/administration/users/user_handler.py

```diff
@@ -62,7 +62,7 @@ def add(self, users_csv_path: pathlib.Path) -> None:
             # Commit changes
             self.active_directory_users.add(users)
         except Exception as exc:
-            msg = f"Could not add users from '{users_csv_path}'.\n{exc!s}"
```

Comment:
Won't you get `exception.__repr__` instead of `exception.__str__` by doing this?

Reply:
Nope, the default conversion is `str()`.